cli/cmd/plugin/unmanaged-cluster/cluster/kind.go

```diff
@@ -158,6 +163,64 @@ func (kcm KindClusterManager) Prepare(c *config.UnmanagedClusterConfig) error {
 	return nil
 }
 
+// PreflightCheck performs any pre-checks that can find issues up front that
+// would cause problems for cluster creation.
+func (kcm KindClusterManager) PreflightCheck() []error {
+	// Check presence of docker
+	cmd := exec.Command("docker", "ps")
+	if err := cmd.Run(); err != nil {
+		// In this case we can't check the rest of the settings, so just return
+		// the one error.
+		return []error{errors.New("docker CLI not found or invalid permissions")}
+	}
+
+	// Get Docker info
+	cmd = exec.Command("docker", "info")
```

Comment:
I'm surprised this would work for your later `json.Unmarshal` call.

Any chance you meant:

```
docker info --format '{{ json . }}'
```

?

Reply:
Oops, that was missed somehow. Good catch!